OCPDOCS_15205: Draft_PR_conf_vms_aws (#111331)
openshift/openshift-docs: modules/installation-aws-enabling-confidential-vms.adoc

// Module included in the following assemblies:
//
// * installing/installing_aws/installing-aws-customizations.adoc
// * installing/installing_aws/installing-aws-network-customizations.adoc
// * installing/installing_aws/installing-aws-private.adoc
// * installing/installing_aws/installing-aws-vpc.adoc
// * installing/installing_aws/installing-restricted-networks-aws-installer-provisioned.adoc
// * installing/installing_aws/installing-aws-user-infra.adoc
// * installing/installing_aws/installing-aws-user-infra-vpc.adoc
// * installing/installing_aws/installing-restricted-networks-aws.adoc

:_mod-docs-content-type: PROCEDURE
[id="installation-aws-enabling-confidential-vms_{context}"]

> = Enabling Confidential VMs with AMD SEV-SNP for {aws-short}

Reviewer: ```suggestion
= Enabling Confidential VMs with AMD SEV-SNP
```

> You can use Confidential VMs with AMD Secure Encrypted Virtualization-Secure Nested Paging (AMD SEV-SNP) when installing your cluster. AMD SEV-SNP is a CPU feature that provides memory encryption and integrity protection for virtual machines (VMs). Confidential VMs encrypt data while it is being processed. For more information, see {aws-first} documentation on link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/sev-snp.html[AMD SEV-SNP].

Reviewer: ```suggestion
[role="_abstract"]
You can use Confidential VMs with AMD Secure Encrypted Virtualization-Secure Nested Paging (AMD SEV-SNP) when installing your cluster. AMD SEV-SNP is a CPU feature that provides memory encryption and integrity protection for virtual machines (VMs). Confidential VMs encrypt data while it is being processed. For more information, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/sev-snp.html[AMD SEV-SNP ({aws-short} documentation)].
```

[NOTE]
====
Confidential VMs are currently not supported on 64-bit ARM architectures.
====

.Procedure

* Use a text editor to edit the `install-config.yaml` file prior to deploying your cluster and add one of the following stanzas:
.. To use confidential VMs for only control plane machines:
+
[source,yaml]
----
controlPlane:
  platform:
    aws:
       confidentialCompute: AMDEncryptedVirtualizationNestedPaging

>        type: r6a.xlarge

Reviewer: ```suggestion
controlPlane:
  platform:
    aws:
      cpuOptions:
        confidentialCompute: AMDEncryptedVirtualizationNestedPaging
```
Updated the parameter to reflect https://github.com/openshift/installer/pull/10012/changes

Reviewer: Did you find something in the feature that specified that a certain instance type is required for AWS confidential VMs, or did you base that off of GCP?

----
+

> Where: 

Reviewer: ```suggestion
where: 
```

> `AMDEncryptedVirtualizationNestedPaging`:: Enable confidential VMs with AMD SEV-SNP. For more information about available options, see "Additional {aws-short} configuration parameters".

Reviewer: ```suggestion
`controlPlane.platform.aws.cpuOptions.confidentialCompute`:: Enables confidential VMs for control plane machines. Valid values are `Disabled` and `AMDEncryptedVirtualizationNestedPaging`.
```

> `r6a.xlarge`:: Specify a instance type that supports Confidential VMs. Confidential VMs require the m6a, c6a, and r6a series of instance types. For more information on supported instance types, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/snp-find-instance-types.html[Instance types that support AMD SEV-SNP].

Reviewer: ```suggestion
`controlplane.platform.aws.type`:: Specifies a instance type that supports Confidential VMs. Confidential VMs require the m6a, c6a, or r6a series of instance types. For more information on supported instance types, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/snp-find-instance-types.html[Instance types that support AMD SEV-SNP].
```

+
.. To use confidential VMs for only compute machines:
+
[source,yaml]
----
compute:
- platform:
    aws:
       confidentialCompute: AMDEncryptedVirtualizationNestedPaging
       type: c6a.xlarge
----

> +

Reviewer: Add similar description lists to the 2nd and 3rd code blocks as you did for the 1st code block.
